tests/benchmarks/peripheral_stress_test/src/pwm_thread.c

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pwm, LOG_LEVEL_INF);
+
+#include <zephyr/drivers/pwm.h>
+#include "common.h"
+
+static const struct pwm_dt_spec pwm_led0 = PWM_DT_SPEC_GET(DT_ALIAS(pwm_led0));
+
+/* PWM thread */
+static void pwm_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	int ret;
+	bool dir_up;
+	uint32_t pwm_period;
+	uint32_t pulse_min;
+	uint32_t pulse_max;
+	int32_t pulse_step, pulse_step_min, pulse_step_diff;
+	uint32_t current_pulse_width;
+
+	atomic_inc(&started_threads);
+
+	if (!pwm_is_ready_dt(&pwm_led0)) {
+		LOG_ERR("Device %s is not ready.", pwm_led0.dev->name);
+		atomic_inc(&completed_threads);
```

Comment:
Not needed here? missing increment will allow to detect that something has failed

Reply:
Failure detected based on LOG_INF("PWM thread has completed");